File: CPlusPlus/CppLibrary/System.cpp
#include "System.h"

#include <iostream>

/**
 * Prints a boolean value.
 *
 * @param b The boolean to be printed
 */
void PrintStream::print(bool b) {
    std::cout << (b ? "true" : "false");
}

/**
 * Prints a character.
 *
 * @param c The char to be printed
 */
void PrintStream::print(char c) {
    std::cout << c;
}

/**
 * Prints an array of characters.
 *
 * @param s The array of chars to be printed
 */
void PrintStream::print(char* s, int32_t length) {
    for (int i = 0; i < length; i++) {
        std::cout << s[i];
    }
}

/**
 * Prints a double-precision floating-point number.
 *
 * @param d The double to be printed
 */
void PrintStream::print(double d) {
    std::cout << d;
}

/**
 * Prints a floating-point number.
 *
 * @param f The float to be printed
 */
void PrintStream::print(float f) {
    std::cout << f;
}

/**
 * Prints an integer.
 *
 * @param i The int to be printed
 */
void PrintStream::print(int32_t i) {
    std::cout << i;
}

/**
 * Prints a long integer.
 *
 * @param l The long to be printed
 */
void PrintStream::print(int64_t l) {
    std::cout << l;
}

/**
 * Prints an object.
 *
 * @param obj The Object to be printed
 */
void PrintStream::print(Object *obj) {
    String *s = obj->toString();
    std::cout << s->_s();
    delete s;
}

/**
 * Prints a string.
 *
 * @param s The String to be printed
 */
void PrintStream::print(String *s) {
    std::cout << s->_s();
}

/**
 * Terminates the current line by writing the line separator string.
 */
void PrintStream::println() {
    std::cout << std::endl;
}

/**
 * Prints a boolean and then terminate the line.
 *
 * @param x The boolean to be printed
 */
void PrintStream::println(bool x) {
    std::cout << (x ? "true" : "false") << std::endl;
}

/**
 * Prints a character and then terminate the line.
 *
 * @param x The char to be printed.
 */
void PrintStream::println(char x) {
    std::cout << x << std::endl;
}

/**
 * Prints an array of characters and then terminate the line.
 *
 * @param x an array of chars to print.
 */
void PrintStream::println(char* x, int32_t length) {
    for (int i = 0; i < length; i++) {
        std::cout << x[i];
    }
    
    std::cout << "" << std::endl;
}

/**
 * Prints a double and then terminate the line.
 *
 * @param x The double to be printed.
 */
void PrintStream::println(double x) {
    std::cout << x << std::endl;
}

/**
 * Prints a float and then terminate the line.
 *
 * @param x The float to be printed.
 */
void PrintStream::println(float x) {
    std::cout << x << std::endl;
}

/**
 * Prints an integer and then terminate the line.
 *
 * @param x The int to be printed.
 */
void PrintStream::println(int32_t x) {
    std::cout << x << std::endl;
}

/**
 * Prints a long and then terminate the line.
 *
 * @param x The long to be printed.
 */
void PrintStream::println(int64_t x) {
    std::cout << x << std::endl;
}

/**
 * Prints an Object and then terminate the line.
 *
 * @param x The Object to be printed.
 */
void PrintStream::println(Object *obj) {
    String *s = obj->toString();
    std::cout << s->_s() << std::endl;
    delete s;
}

/**
 * Prints a String and then terminate the line.
 *
 * @param x The String to be printed.
 */
void PrintStream::println(String *x) {
    std::cout << x->_s() << std::endl;
}

InputStream *System::in = new InputStream();
PrintStream *System::out = new PrintStream();

File: CSharp/CsLibrary/System_.cs
using System;
using System.Threading;

namespace JavaClasses
{

    /// <summary>
    /// A PrintStream adds functionality to another output stream,
    /// namely the ability to print representations of various data values conveniently.
    /// </summary>
    /// <author>Petr Kozler (A13B0359P)</author>
    public class PrintStream : Object
    {

        public PrintStream()
        {
            // setting the standard output format
            Thread.CurrentThread.CurrentCulture = System.Globalization.CultureInfo.InvariantCulture;
        }

        /// <summary>
        /// Prints a boolean value.
        /// </summary>
        /// <param name="b">The boolean to be printed
        /// </param>
        public void print(bool b)
        {
            Console.Write(b);
        }

        /// <summary>
        /// Prints a character.
        /// </summary>
        /// <param name="c">The char to be printed
        /// </param>
        public void print(char c)
        {
            Console.Write(c);
        }

        /// <summary>
        /// Prints an array of characters.
        /// </summary>
        /// <param name="s">The array of chars to be printed
        /// </param>
        public void print(char[] s)
        {
            Console.Write(s);
        }

        /// <summary>
        /// Prints a double-precision floating-point number.
        /// </summary>
        /// <param name="d">The double to be printed
        /// </param>
        public void print(double d)
        {
            Console.Write(d);
        }

        /// <summary>
        /// Prints a floating-point number.
        /// </summary>
        /// <param name="f">The float to be printed
        /// </param>
        public void print(float f)
        {
            Console.Write(f);
        }

        /// <summary>
        /// Prints an integer.
        /// </summary>
        /// <param name="i">The int to be printed
        /// </param>
        public void print(int i)
        {
            Console.Write(i);
        }

        /// <summary>
        /// Prints a long integer.
        /// </summary>
        /// <param name="l">The long to be printed
        /// </param>
        public void print(long l)
        {
            Console.Write(l);
        }

        /// <summary>
        /// Prints an object.
        /// </summary>
        /// <param name="obj">The Object to be printed
        /// </param>
        public void print(Object obj)
        {
            Console.Write(obj);
        }

        /// <summary>
        /// Prints a string.
        /// </summary>
        /// <param name="s">The String to be printed
        /// </param>
        public void print(String s)
        {
            Console.Write(s);
        }

        /// <summary>
        /// Terminates the current line by writing the line separator string.
        public void println()
        {
            Console.WriteLine();
        }

        /// <summary>
        /// Prints a boolean and then terminate the line.
        /// </summary>
        /// <param name="x">The boolean to be printed
        /// </param>
        public void println(bool x)
        {
            Console.WriteLine(x);
        }

        /// <summary>
        /// Prints a character and then terminate the line.
        /// </summary>
        /// <param name="x">The char to be printed.
        /// </param>
        public void println(char x)
        {
            Console.WriteLine(x);
        }

        /// <summary>
        /// Prints an array of characters and then terminate the line.
        /// </summary>
        /// <param name="x">an array of chars to print.
        /// </param>
        public void println(char[] x)
        {
            Console.WriteLine(x);
        }

        /// <summary>
        /// Prints a double and then terminate the line.
        /// </summary>
        /// <param name="x">The double to be printed.
        /// </param>
        public void println(double x)
        {
            Console.WriteLine(x);
        }

        /// <summary>
        /// Prints a float and then terminate the line.
        /// </summary>
        /// <param name="x">The float to be printed.
        /// </param>
        public void println(float x)
        {
            Console.WriteLine(x);
        }

        /// <summary>
        /// Prints an integer and then terminate the line.
        /// </summary>
        /// <param name="x">The int to be printed.
        /// </param>
        public void println(int x)
        {
            Console.WriteLine(x);
        }

        /// <summary>
        /// Prints a long and then terminate the line.
        /// </summary>
        /// <param name="x">The long to be printed.
        /// </param>
        public void println(long x)
        {
            Console.WriteLine(x);
        }

        /// <summary>
        /// Prints an Object and then terminate the line.
        /// </summary>
        /// <param name="x">The Object to be printed.
        /// </param>
        public void println(Object x)
        {
            Console.WriteLine(x);
        }

        /// <summary>
        /// Prints a String and then terminate the line.
        /// </summary>
        /// <param name="x">The String to be printed.
        /// </param>
        public void println(String x)
        {
            Console.WriteLine(x);
        }

    }

    /// <summary>
    /// This abstract class is the superclass of all classes 
    /// representing an input stream of bytes.
    /// </summary>
    /// <author>Petr Kozler (A13B0359P)</author>
    public class InputStream : Object
    {
        // really no code
    }

    /// <summary>
    /// The System class contains several useful class fields and methods.
    /// </summary>
    /// <author>Petr Kozler (A13B0359P)</author>
    public class System_ : Object
    {

        /// <summary>
        /// The "standard" input stream.
        /// </summary>
        public readonly static InputStream in_ = new InputStream();

        /// <summary>
        /// The "standard" output stream.
        /// </summary>
        public readonly static PrintStream out_ = new PrintStream();

    }

}
